src/Mvc/Mvc.ViewFeatures/src/RazorComponents/ComponentRenderer.cs

```diff
@@ -0,0 +1,110 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Components;
+using Microsoft.AspNetCore.Html;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc.Rendering;
+
+namespace Microsoft.AspNetCore.Mvc.ViewFeatures.RazorComponents
```

Comment:
do we need this new namespace? `.ViewFeatures` is already kinda a junk namespace.

Reply:
Yeah, we could clean it up. I thought everything else in that directory was intentionally under the namespace, but it's just this type: https://github.com/aspnet/AspNetCore/blob/master/src/Mvc/Mvc.ViewFeatures/src/RazorComponents/StaticComponentRenderer.cs#L18